addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -321,23 +321,20 @@ def _cron_receive_fattura_pa(self):
                     # should not happen as the file has been checked by SdiCoop
                     _logger.info('Received file badly formatted, skipping: \n %s', file)
                     continue
-                invoice = self.env['account.move'].with_company(company).create({'move_type': 'in_invoice'})
+
                 attachment = self.env['ir.attachment'].create({
                     'name': fattura['filename'],
                     'raw': file,
                     'type': 'binary',
-                    'res_model': 'account.move',
-                    'res_id': invoice.id
                 })
+
                 if not self.env.context.get('test_skip_commit'):
                     self.env.cr.commit() # In case something fails after, we still have the attachment
-                # So that we don't delete the attachment when deleting the invoice
-                attachment.res_id = False
-                attachment.res_model = False
-                invoice.unlink()
-                invoice = self.env.ref('l10n_it_edi.edi_fatturaPA')._create_invoice_from_xml_tree(fattura['filename'], tree)
-                attachment.write({'res_model': 'account.move',
-                                  'res_id': invoice.id})
+
+                self.env['account.journal'].with_company(company)\
```

Comment:
What exactly is the scope of this PR? Looks like you're touching things we're already changing in a lot of current bugfixes and refactorings.   See #113666  #111365 #110249 #113421 #113079

Reply:
Hello @lordkrandel note that we are targeting master but you are right we made some change for Italy as well, mainly to ensure files keep being imported there as well given the new mechanism.

Reply:
Yes, but you're also making non-relevant changes like this one ☝🏻  on the cron, while we are working actively to bugprove on v14 for law requirements, making it harder and harder to merge fw-ports.
I.e. in a rather urgent PR we have to split our current cron in 2 also in v14, one for automatically sending invoices and one to sync the status of the sent invoices with our IAP proxy.

This PR is quite big, and it's hard to see if everything fits and where.
So I'd like the diff to be more minimal, and split these unnecessary changes in a new PR, so that they can be better understood, backported etc.
I really like the approach and the refactorings on the tests really make sense. But I'd like to see its shape more.

Reply:
@smetl @tdetry You removed the fail safe.  As this is triggered through a cron, it can not pop up a user error in case it fails, so in the worst case it would just do an empty invoice with the attachment...